domain/src/Aggregates/SharePoolingAsset/Actions/AcquireSharePoolingAsset.php

```diff
@@ -16,6 +17,8 @@ public function __construct(
         public LocalDate $date,
         public Quantity $quantity,
         public FiatAmount $costBasis,
+        // Testing purposes only
+        public ?SharePoolingAssetTransactionId $id = null,
```

Comment:
ℹ️ Not proud of this. In an ideal world the ID would be generated out of the object's properties instead, but these are not unique enough and there would almost certainly be ID collisions occurring.

This allows me to generate and pass an ID beforehand (rather than let the object generate it for itself) that I can then use for checks within some tests.